smalltalksrc/Slang/TMethod.class.st

```diff
@@ -573,6 +573,15 @@ TMethod >> children [
 	^ { parseTree }
 ]
 
+{ #category : 'inlining support' }
+TMethod >> cleanASTOf: aNode [
+	"supressing an unused variable in a conditional statement can leave it empty. We can supress the conditional branch if we encounter such cases"
+	aNode parent isTMethod ifTrue: [ ^self ].
+	aNode parent remove: aNode.
+	aNode parent children isEmpty ifTrue: [
+		self cleanASTOf: aNode parent ]
+]
```

Comment:
Why is this in TMethod? `self` does not play any role in here, right?

Reply:
Maybe you can name it like `removeUnusedNodesInBranch` and explain in the comment the recursive nature.